src/platform/platform_win.cpp

```diff
@@ -507,52 +526,339 @@ BOOL signal_handler(DWORD dwCtrlType)
+            return str;
+        }
+
+        /**
+         * Convert IPv6 address to string
+         *
+         * @param [in] addr IPv6 address
+         * @return std::string String representation of @p addr
+         */
+        static std::string to_string(const in6_addr& addr)
+        {
+            char str[INET6_ADDRSTRLEN] = {};
+            if (!inet_ntop(AF_INET6, &addr, str, sizeof(str)))
+                throw std::runtime_error("inet_ntop failed: errno");
+            return str;
+        }
+
+        /**
+         * Convert an IPv4 address to network CIDR
+         *
+         * @param [in] v4 IPv4 address
+         * @param [in] prefix_length Network prefix
+         * @return std::string Network address in CIDR form
+         */
+        static auto to_network(const in_addr& v4, std::uint8_t prefix_length)
+        {
+            // Convert to the host long first so we can apply a mask to it
+            constexpr static auto kMaxPrefixLength = 32;
+            const auto ip_hbo = ntohl(v4.S_un.S_addr);
+            if (prefix_length > kMaxPrefixLength)
+            {
+                throw std::runtime_error{"Given prefix length `{}` is larger than `{}`!"};
+            }
+            const auto mask = (prefix_length == 0) ? 0
+                                                   : std::numeric_limits<std::uint32_t>::max()
+                                                         << (32 - prefix_length);
+            const auto network_hbo = htonl(ip_hbo & mask);
+
+            return fmt::format("{}/{}", to_string(network_hbo), prefix_length);
+        }
+
+        /**
+         * Convert an IPv6 address to network CIDR
+         *
+         * @param [in] v6 IPv6 address
+         * @param [in] prefix_length Network prefix
+         * @return std::string Network address in CIDR form
+         */
+        static auto to_network(const in6_addr& v6, std::uint8_t prefix_length)
+        {
+            // Convert to the host long first so we can apply a mask to it
+            constexpr static auto kMaxPrefixLength = 128;
+            if (prefix_length > kMaxPrefixLength)
+            {
+                throw std::runtime_error{"Given prefix length `{}` is larger than `{}`!"};
+            }
+            in6_addr masked = v6;
+
+            for (int i = 0; i < 16; ++i)
+            {
+                int bits = i * 8;
+                if (prefix_length < bits)
+                    masked.u.Byte[i] = 0;
+                else if (prefix_length < bits + 8)
+                    masked.u.Byte[i] &= static_cast<uint8_t>(0xFF << (8 - (prefix_length - bits)));
+            }
+            const auto network_addr = to_string(masked);
+            return fmt::format("{}/{}", network_addr, prefix_length);
+        }
+    } static helper;
+
+    // Initialize once and reuse.
+    return helper;
+}
+
 std::map<std::string, mp::NetworkInterfaceInfo>
 mp::platform::Platform::get_network_interfaces_info() const
```

Comment:
Idem, large function would benefit from being split into smaller chunks. However, isn't the previous PS command available regardless of Hyper-V? If so, why reimplement?